packages/x-data-grid/src/tests/rows.DataGrid.test.tsx

```diff
@@ -745,7 +745,9 @@ describe('<DataGrid /> - Rows', () => {
             width={100}
           />,
         );
-        expect($$(`.${gridClasses.cell}:not(.${gridClasses.cellEmpty})`)).to.have.length(2);
+        await waitFor(() => {
```

Comment:
Is `waitFor` necessary here? Was the test failing after applying the fix?

Reply:
> Was the test failing after applying the fix?

Yep https://app.circleci.com/pipelines/github/mui/mui-x/82116/workflows/04046e3d-35d3-4676-93a4-de6436aca75b/jobs/467454

I didn't quite get to the root of the problem, but noticed that on first render the `rowHasAutoHeight` function here returns `false` https://github.com/mui/mui-x/blob/master/packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx#L882 but then corrects itself on second render. 

I don't know if the `waitFor` is masking a bigger issue but no other tests failed 🤷